### ISSUE TITLE
ZTEX 1.15y (not the rev2) issue

### ISSUE THREAD

**Issue by blandyuk** [openwall/john#4710]
Hi,

I have 6 ZTEX boards, 5 are the 1.15y rev2 and one is the older 1.15y only.

The 5x rev2 boards work OK, but the 1.15y board does the below:

SN 04A3465A70: firmware uploaded
SN: 04A3465A70 productId: 10.15.0.0 "inouttraffic 1.0.0 JtR" busnum:2 devnum:6
Using default input encoding: UTF-8
Loaded 1 password hash (bcrypt-ztex [Blowfish ZTEX])
Cost 1 (iteration count) is 1024 for all loaded hashes
SN 04A3465A70: uploading bitstreams.. ok
SN 04A3465A70: device_list_check_bitstreams(): no bitstream or wrong type

I've uploaded numerous versions of the default firmware from official ztex.de website for 1.15y to no avail, but have read that it might just be john not liking this older board maybe:

https://www.openwall.com/lists/john-dev/2016/11/14/2

Any help would be, thanks.

**Comment by solardiz** [comment]
That john-dev posting you reference does explain what the error message means, but the rest of it is mostly out of date. Also, no need for you to be uploading firmware manually, as the excerpt above shows JtR replaces the firmware just fine (which it does without making that permanent, which is also fine).

I suggest you look for a hardware issue.

Try changing the USB hub, port, cable. Is the USB link possibly operating at low speed? This is known to be unreliable with these boards. So if there's an old or cheap USB hub in the way that only does USB 1.1, remove it.

Is the board getting sufficient power with no interruptions?

In my experience, JtR works with pre-rev2 boards fine. In fact, I think none of mine are rev2. I just took a look at one - it's marked "1.15y" and that's it.

**Comment by blandyuk** [comment]
Thanks as always solardiz.

Yes, done all the usual, changed USB cable (using short ones as they work better), different power inputs. Connected direct so no USB hub for this one. The other 5x are connected via 10 port USB3 hub and working perfect.

Will be a hardware problem, so that's that. Originally, the board started with the "SN 04A3465A70" but once john uploaded firmware, it changed to "SN 0000001337". Presume this relates to an error code, maybe.

I have 5 of them, so that will have to do for now until I can get more somehow.

Thanks again for the help.

**Comment by solardiz** [comment]
> Will be a hardware problem, so that's that.

We used to test these boards with a certain older version of cgminer, from just before they dropped the support, trying to mine Bitcoin. I think this is still supposed to work. You can try it with your boards.

> Originally, the board started with the "SN 04A3465A70" but once john uploaded firmware, it changed to "SN 0000001337".

I guess it's the other way around. In fact, you show that real-looking SN in your first comment here, suggesting it's the one seen after firmware upload (and that `john` had already uploaded its firmware into that board prior to the copy-paste).

> Presume this relates to an error code, maybe.

No, this suggests that whatever firmware you have in EEPROM has that in there. I've also seen something like this with a certain board - the SN changes from fake to real upon JtR's firmware upload. That board works just fine.

**Comment by blandyuk** [comment]
OK thanks. There are tools from the ztex downloads which might allow me to resolve this? We have the FWLoader and the DeviceServer of which I've tried both so know how they work.

Cannot remember if I can change what's in the EEPROM. I know DeviceServer has numerous options. I have numerous downloads of different versions for the SDK, any you recommend using?

https://www.ztex.de/downloads/
See: ZTEX EZ-USB SDK

The olders ones have the ztex 1.15y. The latest I found was: ztex-170322
Older one is: 140813

**Comment by solardiz** [comment]
I have no recommendations on those tools. Maybe @Apingis or someone on john-users does (after all, this is more of a john-users than a GitHub topic).

My recommendation is that you test with sufficiently old cgminer.

**Comment by solardiz** [comment]
> Cannot remember if I can change what's in the EEPROM.

Maybe I wasn't clear. Let me try again: I think you don't need to change what's in EEPROM.

**Comment by blandyuk** [comment]
OK, I've tested with cgminer as suggested, and it looks like hw error. Pastebin or benchmark is below:

https://pastebin.com/hYsQtMui

Thanks for all the help, but looks like the board is a paper-weight.

**Comment by solardiz** [comment]
To make sure your cgminer setup is right, also test one of the known-working boards with it.

BTW, you shouldn't need to run these programs under `sudo` if you configure `udev` rules as described in our `README-ZTEX`.

**Comment by blandyuk** [comment]
OK cheers. The other boards are busy, but I'll test at some point.
